### PR TITLE
Add indicator feature

### PR DESCRIPTION
**Pull request by derenrich**
Add support for indicator features (1 if a feature is present / 0 otherwise)

### CODE REVIEW

**Comment by regadas** [comment]
Yeah I’ll have a look

File: core/src/main/scala/com/spotify/featran/transformers/Indicator.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 Spotify AB.
```
Comment (regadas) [inline core/src/main/scala/com/spotify/featran/transformers/Indicator.scala]: can you update to 2019?

File: core/src/test/scala/com/spotify/featran/Fixtures.scala [hunk 2]
```diff
@@ -88,6 +88,7 @@ object Fixtures {
     .required(_.s3)(HashNHotWeightedEncoder("hash-n-hot-weighted"))
     .required(_.s1)(HeavyHitters("heavy-hitters", 10, 0.001, 0.001, 1))
     .required(_.x)(Identity("id"))
+    .optional(_.xo)(Indicator("indic"))
```
Comment (regadas) [inline core/src/test/scala/com/spotify/featran/Fixtures.scala]: why not have the full name?

File: core/src/test/scala/com/spotify/featran/transformers/IndicatorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017 Spotify AB.
```
Comment (regadas) [inline core/src/test/scala/com/spotify/featran/transformers/IndicatorSpec.scala]: 🎨 2019

File: core/src/main/scala/com/spotify/featran/transformers/Indicator.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  /**
+   * Create a new [[Indicator]] from a settings object
+   * @param setting Settings object
+   */
+  def fromSettings(setting: Settings): Transformer[Double, Unit, Unit] =
+    Indicator(setting.name)
+}
+
+private[featran] class Indicator(name: String) extends OneDimensional[Double, Unit, Unit](name) {
```
Comment (regadas) [inline core/src/main/scala/com/spotify/featran/transformers/Indicator.scala]: you could potentially just use `MapOne` and override the map function

```scala
class Indicator(name: String) extends MapOne[Double](name) {
  override def map(a: Double): Double = 1

  override def flatRead[T: FlatReader]: T => Option[Any] = FlatReader[T].readDouble(name)

  override def flatWriter[T](implicit fw: FlatWriter[T]): Option[Double] => fw.IF =
    fw.writeDouble(name)
}
```

Comment (derenrich) [reply core/src/main/scala/com/spotify/featran/transformers/Indicator.scala]: this is ... a good point

**Comment by derenrich** [comment]
comments addressed

**Pull request merged**